gui/wxpython/datacatalog/tree.py

```diff
@@ -688,67 +719,73 @@ def OnPasteMap(self, event):
 
         for i in range(len(self.copy_layer)):
             gisrc, env = gscript.create_environment(
-                            gisenv()['GISDBASE'], self.selected_location[0].label, mapset=self.selected_mapset[0].label)
+                            self.selected_grassdb[0].label,
+                            self.selected_location[0].label,
+                            self.selected_mapset[0].label)
             gisrc2, env2 = gscript.create_environment(
-                            gisenv()['GISDBASE'], self.copy_location[i].label, mapset=self.copy_mapset[i].label)
+                            self.selected_grassdb[i].label,
+                            self.copy_location[i].label,
+                            self.copy_mapset[i].label)
             new_name = self.copy_layer[i].label
-            if self.selected_location[0] == self.copy_location[i]:
-                # within one mapset
-                if self.selected_mapset[0] == self.copy_mapset[i]:
-                    # ignore when just moves map
-                    if self.copy_mode is False:
-                        return
-                    new_name = self._getNewMapName(_('New name for <{n}>').format(n=self.copy_layer[i].label),
-                                                   _('Select new name'),
-                                                   self.copy_layer[i].label, env=env,
-                                                   mapset=self.selected_mapset[0].label,
-                                                   element=self.copy_layer[i].data['type'])
-                    if not new_name:
-                        return
-                # within one location, different mapsets
-                else:
-                    if map_exists(new_name, element=self.copy_layer[i].data['type'], env=env,
-                                  mapset=self.selected_mapset[0].label):
+
+            if self.selected_grassdb[0] == self.copy_grassdb[i]:
+                if self.selected_location[0] == self.copy_location[i]:
```

Comment:
The point of the condition here is to test if you are copying/moving map in a different location, in that case you need to reproject. If it's within the same location you don't need to reproject. If the location is in a different database, that doesn't make difference, you still need to reproject. So just remove that test you added.